virttest/libvirt_xml/vm_xml.py

```diff
@@ -538,21 +538,30 @@ def del_controller(self, controller_type=None):
 
         :return: None if deleting all controllers
         """
-        all_controllers = self.xmltreefile.findall("devices/controller")
-        del_controllers = []
-        for controller in all_controllers:
-            if controller.get("type") != controller_type:
-                continue
-            del_controllers.append(controller)
-
         # no seclabel tag found in xml.
+        del_controllers = self.get_controllers()
         if del_controllers == []:
             logging.debug("Controller %s for this domain does not "
                           "exist" % controller_type)
 
         for controller in del_controllers:
             self.xmltreefile.remove(controller)
 
+    def get_controllers(self, controller_type=None):
+        """
+        Get controllers according controller type
+
+        :param controller_type: type of controllers need to get
+        :return: controller list
+        """
+        all_controllers = self.xmltreefile.findall("devices/controller")
+        type_controllers = []
+        for controller in type_controllers:
```

Comment:
why do we iterate empty list here, we should be using `all_controllers`

Reply:
fixed, thanks